ddtrace/contrib/internal/azure_eventhubs/patch.py

```diff
@@ -0,0 +1,219 @@
+        _w("azure.eventhub.aio", "EventHubProducerClient._buffered_send_batch", _patched_send_batch_async)
+    else:
+        Pin().onto(azure_eventhubs_module.EventHubProducerClient)
+        Pin().onto(azure_eventhubs_module.EventDataBatch)
+        _w("azure.eventhub", "EventDataBatch.add", _patched_add)
+        _w("azure.eventhub", "EventHubProducerClient.__init__", _patched_producer_init)
+        _w("azure.eventhub", "EventHubProducerClient.create_batch", _patched_create_batch)
+        _w("azure.eventhub", "EventHubProducerClient.send_event", _patched_send_event)
+        _w("azure.eventhub", "EventHubProducerClient.send_batch", _patched_send_batch)
+        _w("azure.eventhub", "EventHubProducerClient._buffered_send_event", _patched_send_event)
+        _w("azure.eventhub", "EventHubProducerClient._buffered_send_batch", _patched_send_batch)
+
+
+def _patched_producer_init(wrapped, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return wrapped(*args, **kwargs)
+
+    instance._dd_fully_qualified_namespace = get_argument_value(args, kwargs, 0, "fully_qualified_namespace", True)
+
+    return wrapped(*args, **kwargs)
+
+
+def _patched_create_batch(wrapped, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled() or not config.azure_eventhubs.batch_links:
+        return wrapped(*args, **kwargs)
+
+    batch = wrapped(*args, **kwargs)
+
+    batch._dd_eventhub_name = instance.eventhub_name
+    batch._dd_fully_qualified_namespace = instance._dd_fully_qualified_namespace
+
+    return batch
+
+
+async def _patched_create_batch_async(wrapped, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled() or not config.azure_eventhubs.batch_links:
+        return await wrapped(*args, **kwargs)
+
+    batch = await wrapped(*args, **kwargs)
+
+    batch._dd_eventhub_name = instance.eventhub_name
+    batch._dd_fully_qualified_namespace = instance._dd_fully_qualified_namespace
+
+    return batch
+
+
+def _patched_add(wrapped, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if (
+        not pin
+        or not pin.enabled()
+        or not config.azure_eventhubs.batch_links
+        # Skip patching when these attributes haven't been added.
+        # A known case is when a producer client in buffered mode
+        # instantiates a batch without using the create_batch method.
+        or not getattr(instance, "_dd_eventhub_name", None)
+        or not getattr(instance, "_dd_fully_qualified_namespace", None)
+    ):
+        return wrapped(*args, **kwargs)
+
+    resource_name = instance._dd_eventhub_name
+    fully_qualified_namespace = instance._dd_fully_qualified_namespace
+    operation_name = f"{azure_eventhubsx.CLOUD}.{azure_eventhubsx.SERVICE}.{azure_eventhubsx.CREATE}"
+
+    with create_context("azure.eventhub.patched_producer_batch", pin, operation_name, resource_name) as ctx, ctx.span:
```

Comment:
```suggestion
    with create_context("azure.eventhub.patched_producer_batch", pin, operation_name, resource_name) as ctx:
```

We want to deprecate `ctx.span` usage

if you add `"azure.functions.patched_event_hubs"` and etc to the list of functions that we call `_finish_span` on, then we don't need `ctx.span` here at all (and in the other places)

which it doesn't look like you directly need the span in any of these wrappers anyways.